packages/compass-indexes/src/components/indexes-table/indexes-table.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useMemo } from 'react';
+import {
+  css,
+  Table,
+  TableHeader,
+  Row,
+  Cell,
+  cx,
+  spacing,
+} from '@mongodb-js/compass-components';
+
+import NameField from './name-field';
+import TypeField from './type-field';
+import SizeField from './size-field';
+import UsageField from './usage-field';
+import PropertyField from './property-field';
+import DropField from './drop-field';
+
+// When row is hovered, we show the delete button
+const rowStyles = css({
+  ':hover': {
+    '.delete-cell': {
+      visibility: 'visible',
+    },
+  },
+});
+// When row is not hovered, we hide the delete button
+const deletFieldStyles = css({
+  visibility: 'hidden',
```

Comment:
Thinking about keyboard navigation and accessibility here as well, should the delete button be something that a user can navigate to with the keyboard? I'm wondering if it should be able to be tabbed to.